specs/src/specs/header-sync.md

```diff
@@ -0,0 +1,37 @@
+# P2P Header Sync
+
+## Abstract
+
+The P2P Header Sync is a p2pP exchange service for rollkit headers that implements the go-header interface. The main components are:
+
+* store: a `headerEx` prefixed datastore where synced headers are stored
+* subscriber: a libp2p node pubsub subscriber
+* p2p server: a server for handling header requests between peers in the p2p network
+* exchange: a client that enables sending in/out-bound header requests from/to the p2p network
+* syncer: a service for efficient synchronization for headers. When a p2p node falls behind and wants to catch up to the latest network head via p2p network, it can use the syncer.
+
+## Details
+
+All three types of nodes (sequencer, full, and light) run the P2P header exchange service to maintain the cannonical view of the rollup chain (with respect to the p2p network).
+
+The header exchange service inherits the ConnectionGater from the node's p2p client which allows blocking and allowing peers as needed by specifying the `P2PConfig.BlockedPeers` and `P2PConfig.AllowedPeers`.
+
+`NodeConfig.BlockTime` is used to configure the syncer such that it can effectively decide the outdated headers while it receives headers from the p2p network.
+
+Both header and block sync utilizes go-header library and runs two separate exchange services (p2p header sync and p2p block sync). This distinction is mainly to serve light nodes which does not store blocks, but only headers synced from the p2p network. The two separate service may be redundant for the full nodes and can be optimized to single service in the future.
```

Comment:
Question to the team, do we want to include references to future optimizations in the spec? personally I don't think we should because it makes it harder to maintain when thinking about what the trigger would be to update the specs for any potential future case. It is better to keep them focused on current state only.

Open to discussion.

Reply:
Agreed, think it's better to focus on current state

Reply:
sounds good.